torch/tensor.py

```diff
@@ -722,4 +722,25 @@ def _update_names(self, names, inplace):
         else:
             return super(Tensor, self).rename(names)
 
+    @property
+    def grad(self):
+        """
+        This attribute is ``None`` by default and becomes a Tensor the first time a call to
+        :func:`backward` computes gradients for ``self``.
+        The attribute will then contain the gradients computed and future calls to
+        :func:`backward` will accumulate (add) gradients into it.
+        """
+        if self.requires_grad and not hasattr(self, "retains_grad") and not self.is_leaf:
```

Comment:
I don't really understand how setting manually works.  I can always set it manually but if I try to access it when I haven't set requires_grad will warn me?

Reply:
I reduced a lot the scope of the warning.
I will add a `and self._grad is None`.
So that it will only warn when you have a Tensor that requires gradient, is inside the graph, does not retain_grad and has no gradient set.